Extend nameless support for all Avalon monitors and drivers, fixes some deprecations

WernerFS opened the pull request:
Fixes:
- In the class definition, the name parameter is changed for an `Optional[str] = None` for all Avalon drivers and monitors.
- All Avalon tests were corrected to reflect the new parameter order `(entity, name, clock, ...) -> (entity, clock, name, ...)`

Additions:
- Created dedicated test to validate correct interaction with unnamed Avalon ports.
- Tests will only run on Verilator (Icarus does not support interfaces or modports).

Deprecations:
- Substituted `kill` calls in monitor and drivers for `cancel`.
- Substituted `MODULE` for `COCOTB_TEST_MODULES` in all test Makefiles.

ktbarrett commented on tests/test_cases/test_avalon/test_avalon.py:
```diff
@@ -51,8 +51,8 @@ def __init__(self, dut, avlproperties={}):
             dut,
-            "master",
             dut.clk,
+            "master",
```
This breaks people for little reason. We won't accept this change.

ktbarrett reviewed (changes requested):
I say just add the new test.

WernerFS commented:
Updated with the requested changes. I reverted all changes and left only the `name = Optional[str]` to hopefully hint to new users that if they need unnamed instances they can just pass `name = None`. The test can also be used as an example for this functionality. Sorry it took me a while to realize that I didn't need to change the API to use it.

WernerFS commented:
Thanks for the quick reply. I was confident `pre-commit` was already doing its job 😅.

ktbarrett commented:
Closing and reopening to get it to choose a new base with the CI update.

ktbarrett commented:
@WernerFS Looks like there are rebase conflicts. Can you rebase on master?

WernerFS commented:
@ktbarrett Done. Let me know if there is anything else. Honestly I think all this could have been just an entry into the docs stating that for unnamed ports one should pass `name = None`.  Anyways, thanks for bearing with me.

ktbarrett commented:
We use rebase flow, merges do not resolve issues. I tried rebasing and pushing myself and did an oopsie and couldn't fix it because you used your master branch for this PR, so I reuploaded it in #112.
